build-logic/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt

```diff
@@ -96,18 +97,29 @@ abstract class BuildCommitDistribution @Inject internal constructor(
             "-PtoolingApiShadedJarInstallPath=" + commitDistributionToolingApiJar.get().asFile.absolutePath,
             "-PbuildCommitDistribution=true"
         )
-
         if (project.gradle.startParameter.isBuildCacheEnabled) {
             buildCommands.add("--build-cache")
+        }
+
+        return buildCommands.toTypedArray()
+    }
 
+    private
+    fun getBuildEnvs(): Map<String, String> {
+        val envs = mutableMapOf<String, String>()
+
+        if (project.gradle.startParameter.isBuildCacheEnabled) {
             val buildCacheConf = (project.gradle as GradleInternal).settings.buildCache
             val remoteCache = buildCacheConf.remote as HttpBuildCache?
             if (remoteCache?.url != null) {
-                buildCommands.add("-Dgradle.cache.remote.url=${remoteCache.url}")
-                buildCommands.add("-Dgradle.cache.remote.username=${remoteCache.credentials.username}")
+                envs["GRADLE_CACHE_REMOTE_URL"] = remoteCache.url.toString()
+                envs["GRADLE_CACHE_REMOTE_USERNAME"] = remoteCache.credentials.username!!
+                envs["GRADLE_CACHE_REMOTE_PASSWORD"] = remoteCache.credentials.password!!
```

Comment:
IIUC, the password is for pushing? Do we need to pass the `CI` environment variable as well so pushing is on? Or does this happen automatically?